app/src/main/kotlin/com/hippo/ehviewer/util/AppConfig.kt

```diff
@@ -58,17 +66,14 @@ object AppConfig {
 
     fun getTempDir(filename: String): File? = getDirInExternalAppDir(TEMP)?.run { File(this, filename) }
 
-    val externalTempDir: File?
-        get() = appCtx.externalCacheDir?.run { File(this, TEMP).takeIf { it.ensureDirectory() } }
-
+    val externalTempDir
+        get() = appCtx.externalCacheDir?.toOkioPath()?.let { it / TEMP }?.apply { check(ensureDirectory()) }
```

Comment:
Why from return null to throw on failure? Consider using !! to make it non-null if that's intended.

Reply:
Fair.